From ae13765a65be02a058c6b10d525911d34cc98116 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Mon, 4 Sep 2023 14:42:46 +0800
Subject: [PATCH] configs: Enable commands for ostree

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>

Check go after running bootscript

The go is set by ostree's boot.scr, this can make "U-Boot console" menu
stop immediately and go into the u-boot shell, otherwise, it will try other
boot targets and cause wild errors when select "U-Boot console".

Upstream-Status: Inappropriate [ ostree's boot.scr specific ]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 configs/s32g274ardb2_defconfig | 3 +++
 configs/s32g2xxaevb_defconfig  | 3 +++
 configs/s32g399ardb3_defconfig | 3 +++
 configs/s32g3xxaevb_defconfig  | 3 +++
 include/configs/s32-cc.h       | 6 ++++++
 5 files changed, 18 insertions(+)

diff --git a/configs/s32g274ardb2_defconfig b/configs/s32g274ardb2_defconfig
index d6eeb24ace..4054136678 100644
--- a/configs/s32g274ardb2_defconfig
+++ b/configs/s32g274ardb2_defconfig
@@ -1,3 +1,6 @@
 CONFIG_ARM=y
 CONFIG_ARCH_S32G2=y
 CONFIG_SD_BOOT=y
+CONFIG_FAT_WRITE=y
+CONFIG_MENU=y
+CONFIG_CMD_BOOTMENU=y
diff --git a/configs/s32g2xxaevb_defconfig b/configs/s32g2xxaevb_defconfig
index 6d21f0176f..e507a56279 100644
--- a/configs/s32g2xxaevb_defconfig
+++ b/configs/s32g2xxaevb_defconfig
@@ -4,3 +4,6 @@ CONFIG_TARGET_S32G2XXAEVB=y
 CONFIG_SD_BOOT=y
 CONFIG_USB_ULPI_VIEWPORT=y
 CONFIG_S32CC_QSPI_FREQ=166
+CONFIG_FAT_WRITE=y
+CONFIG_MENU=y
+CONFIG_CMD_BOOTMENU=y
diff --git a/configs/s32g399ardb3_defconfig b/configs/s32g399ardb3_defconfig
index 38399e2a6e..1edaca44cb 100644
--- a/configs/s32g399ardb3_defconfig
+++ b/configs/s32g399ardb3_defconfig
@@ -1,3 +1,6 @@
 CONFIG_ARM=y
 CONFIG_ARCH_S32G3=y
 CONFIG_SD_BOOT=y
+CONFIG_FAT_WRITE=y
+CONFIG_MENU=y
+CONFIG_CMD_BOOTMENU=y
diff --git a/configs/s32g3xxaevb_defconfig b/configs/s32g3xxaevb_defconfig
index 269d9f66ea..09be8ad85f 100644
--- a/configs/s32g3xxaevb_defconfig
+++ b/configs/s32g3xxaevb_defconfig
@@ -4,3 +4,6 @@ CONFIG_TARGET_S32G3XXAEVB=y
 CONFIG_SD_BOOT=y
 CONFIG_USB_ULPI_VIEWPORT=y
 CONFIG_S32CC_QSPI_FREQ=166
+CONFIG_FAT_WRITE=y
+CONFIG_MENU=y
+CONFIG_CMD_BOOTMENU=y
diff --git a/include/configs/s32-cc.h b/include/configs/s32-cc.h
index 93865a324e..4ff3126922 100644
--- a/include/configs/s32-cc.h
+++ b/include/configs/s32-cc.h
@@ -241,6 +241,12 @@
 	"mmc dev ${mmcdev}; " \
 	"if mmc rescan; " \
 	"then " \
+		"if run loadbootscript; then " \
+			"run loadfdt; run bootscript; " \
+			"if test ${go} = 0; then " \
+				"exit; " \
+			"fi; " \
+		"fi; " \
 		"if run loadimage; "\
 		"then " \
 			"run mmcboot; " \
-- 
2.25.1

